include/boost/gil/color_convert.hpp

```diff
@@ -150,44 +150,46 @@ struct default_color_converter_impl<rgb_t,gray_t> {
 /// by Burger, Wilhelm, Burge, Mark J.
 /// and it is a gross approximation not precise enough for professional work.
 ///
-/// \todo FIXME: The original implementation did not handle properly signed CMYK pixels as destination
-///
 template <>
 struct default_color_converter_impl<rgb_t, cmyk_t>
 {
     template <typename SrcPixel, typename DstPixel>
     void operator()(SrcPixel const& src, DstPixel& dst) const
     {
         using src_t = typename channel_type<SrcPixel>::type;
-        src_t const r = get_color(src, red_t());  
-        src_t const g = get_color(src, green_t());
-        src_t const b = get_color(src, blue_t());
+        src_t const r  = get_color(src,red_t());
+        src_t const g  = get_color(src,green_t());
+        src_t const b  = get_color(src,blue_t());
 
-        using dst_t   = typename channel_type<DstPixel>::type;
-        dst_t const c = channel_invert(channel_convert<dst_t>(r)); // c = 1 - r
-        dst_t const m = channel_invert(channel_convert<dst_t>(g)); // m = 1 - g
-        dst_t const y = channel_invert(channel_convert<dst_t>(b)); // y = 1 - b
-        dst_t const k = (std::min)(c, (std::min)(m, y));           // k = minimum(c, m, y)
+        using dst_us_t = typename channel_type< cmyk8_pixel_t >::type;
```

Comment:
`us` in `dst_us_t` stands for?

Reply:
us - unsigned

Reply:
Well, I don't think it's a self-descriptive naming.


Moreover, `dst` does no longer fit in that new context as the `dst_us_t` is a (unsigned) type of an intermediate calculation value, and I'd make the intention explicit:

```
using uint_t = typename channel_type<cmyk8_pixel_t>::type;
```

By the way, in GIL, `cmyk8_*` is based on unsigned and `cmyk8s_*` is signed.

Reply:
Understood, thanks.

(also I assume I will have to make some changes to this pull request, and not create another one. )

Reply:
Yes, anything related to this fix should be added to this PR.
Simply, add more commits to your `bugfix/fixed-conversion-of-rgb-to-signed-cmyk-issue#479` branch and just `git push` to your fork (no need to `git push --force`). We will squash this PR before merging, so any fixup and intermediate commits will disappear.

Reply:
Ok, understood.
Thanks again.